week-2-storage/test_parallelization.py

```diff
@@ -0,0 +1,228 @@
+from typing import Dict, List, Optional, Text, Tuple
+import torch
+from transformers import BertModel
+from transformers import BertTokenizer
+import pandas as pd
+import nltk
+from nltk.corpus import stopwords
+import re
+from nltk.stem import WordNetLemmatizer
+import time
+from timeit import default_timer
+import os
+import gdown
+import multiprocessing
+import torch.multiprocessing as mp
+
+
+"""
+Brief Summary:
+Number of cpus                      : 8
+Dataset len                         : 2834
+[Single Process] Preprocessing      : 00:00:05
+[Single Process] Inference          : 00:01:56
+[Single Process] Total              : 00:02:02
+[Multiple Processes] Preprocessing  : 00:00:01
+[Multiple Processes] Inference      : 00:01:52
+[Multiple Processes] Total          : 00:01:53
+
+
+Ideas for text processing were taken from:
+- https://towardsdatascience.com/a-guide-to-cleaning-text-in-python-943356ac86ca
+"""
+
+
+class RegressionModel(torch.nn.Module):
+    """
+    Example of Bert feature extraction:
+    >>> model = BertModel.from_pretrained("bert-base-uncased")
+    >>> tokenizer = BertTokenizer.from_pretrained('bert-base-uncased')
+    >>> text = ["Replace me by any text you'd like.",
+        "Replace me by any text you'd like.", "Replace me by any text you'd like."]
+    >>> encoded_input = tokenizer(text, return_tensors='pt', padding=True)
+    >>> features = model(**encoded_input)
+    >>> features.last_hidden_state.size()
+    """
+
+    def __init__(self) -> None:
+        super(RegressionModel, self).__init__()
+        self.base_model = BertModel.from_pretrained("bert-base-uncased")
+        self.base_model.requires_grad_(False)
+        self.linear = torch.nn.Linear(768, 1)
+        self.linear.requires_grad_(True)
+
+    def forward(self, x: Dict[str, torch.Tensor]) -> torch.Tensor:
+        features = self.base_model(**x)
+        output = torch.mean(features.last_hidden_state, dim=-2)
+        output = self.linear(output)
+        return output
+
+
+def init_model() -> Tuple[torch.nn.Module, torch.nn.Module, torch.device]:
+    if not os.path.exists("./weights"):
+        os.mkdir("./weights")
+    if not os.path.exists("./weights/bert-base-uncased-regression-weights.pth"):
+        gdown.download(
+            fuzzy=True, 
+            url="https://drive.google.com/file/d/1M5SFB5cYS7Q3oLpkEQVGERXtquwPxury/view?usp=sharing",
```

Comment:
would be nice to move this into constants

Reply:
Code updated